src/nullablevector.jl

```diff
@@ -194,8 +185,28 @@ collection to `append!`.
 function Base.append!(X::NullableVector, items::AbstractVector)
     old_length = length(X)
     nitems = length(items)
+    items_copy = convert(typeof(X), items)
     resize!(X, old_length + nitems)
-    X[old_length + 1:end] = items[1:nitems]
+    X[old_length+1:end] = items_copy
+    return X
+end
+
+@doc """
+`prepend!(X::NullableVector, items::AbstractVector)`
+
+Add the elements of `items` to the beginning of `X`.
+
+Note that `prepend!(X, [1, 2, 3])` is equivalent to `unshift!(X, 1, 2, 3)`,
+where the items to be added to `X` are passed individually to `unshift!` and as a
+collection to `prepend!`.
+""" ->
+function Base.prepend!(X::NullableVector, items::AbstractVector)
+    old_length = length(X)
+    nitems = length(items)
+    items_copy = convert(typeof(X), items)
+    ccall(:jl_array_grow_beg, Void, (Any, UInt), X.values, nitems)
```

Comment:
Looks like we need a new function in Julia to allow doing this in pure Julia at some point.